src/network/taraxa_capability.cpp

```diff
@@ -411,7 +411,12 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
         for (auto &block : level_blocks.second) {
           auto hash = blk_hash_t(block);
           if (known_non_finalized_blocks.count(hash) == 0) {
-            dag_blocks.emplace_back(db_->getDagBlock(hash));
+            if (auto blk = db_->getDagBlock(hash); blk) {
+              dag_blocks.emplace_back(blk);
+            } else {
+              assert(false);
```

Comment:
Lets switch this assert with LOG so we actually also get the log message before abort...

Reply:
I do not think we need to, assert is for debug and log for production. I doubt it will be saved somewhere on abort

Reply:
not sure if I get this. Logs are for production as well as for debug. In case we abort here in debug and that abort is not printed in our boost logs, it would be nice to see the error message as last message, no ?

Reply:
abort will kill the node

Reply:
so before killing the node it would be nice to print error message